pkg/vulnsrc/vulnerability/vulnerability.go

```diff
@@ -15,15 +15,38 @@ var (
 		RubySec, RustSec, PhpSecurityAdvisories, NodejsSecurityWg, PythonSafetyDB}
 )
 
-func GetDetail(vulnID string) (types.Severity, string, string, []string) {
-	details, err := db.Config{}.GetVulnerabilityDetail(vulnID)
+// TODO: Until we can dependency inject, we need to monkey patch sadly
+var (
+	getVulnerabilityDetailFunc = db.Config{}.GetVulnerabilityDetail
+)
+
+func GetDetail(vulnID string) (types.Severity, types.VendorSeverity, string, string, []string) {
+	details, err := getVulnerabilityDetailFunc(vulnID)
 	if err != nil {
 		log.Println(err)
-		return types.SeverityUnknown, "", "", nil
+		return types.SeverityUnknown, nil, "", "", nil
 	} else if len(details) == 0 {
-		return types.SeverityUnknown, "", "", nil
+		return types.SeverityUnknown, nil, "", "", nil
+	}
+
+	return getSeverity(details), getVendorSeverity(details), getTitle(details), getDescription(details), getReferences(details)
+}
+
+func getVendorSeverity(details map[string]types.VulnerabilityDetail) types.VendorSeverity {
+	vs := make(types.VendorSeverity)
+	for vendor, detail := range details {
+		switch {
+		case detail.Severity != types.SeverityUnknown:
+			vs[vendor] = detail.Severity
+		case detail.CvssScore != 0:
+			vs[vendor] = scoreToSeverity(detail.CvssScore)
+		case detail.CvssScoreV3 != 0:
+			vs[vendor] = scoreToSeverity(detail.CvssScoreV3)
+		default:
+			vs[vendor] = types.SeverityUnknown
+		}
```

Comment:
Are these lines necessary?

Reply:
It is not, if my understanding is correct we can take NVD if vendor is not present. I'll remove this. https://github.com/aquasecurity/trivy-db/pull/32/commits/d65430e7cb51687599c40063d825235b14ad2373